code/game/objects/structures/__structure.dm

```diff
@@ -145,25 +145,38 @@
 		take_damage(dmg)
 
 /obj/structure/Destroy()
-	reset_mobs_offset()
 	var/turf/T = get_turf(src)
 	if(T)
 		T.fluid_update()
 	. = ..()
+	if(T)
+		for(var/atom/movable/AM in T)
+			AM.reset_offsets()
+			AM.reset_plane_and_layer()
 
-/obj/structure/Crossed(mob/living/M)
-	if(istype(M))
-		M.on_structure_offset(mob_offset)
-	..()
+/obj/structure/Crossed(O)
+	. = ..()
+	if(ismob(O))
+		var/mob/M = O
+		M.reset_offsets()
+		M.reset_plane_and_layer()
 
-/obj/structure/proc/reset_mobs_offset()
-	for(var/mob/living/M in loc)
-		M.on_structure_offset(0)
+/obj/structure/Uncrossed(O)
+	. = ..()
+	if(ismob(O))
+		var/mob/M = O
+		M.reset_offsets()
+		M.reset_plane_and_layer()
 
 /obj/structure/Move()
+	var/turf/T = get_turf(src)
 	. = ..()
 	if(. && !CanFluidPass())
 		fluid_update()
+		if(T)
+			for(var/atom/movable/AM in T)
+				AM.reset_offsets()	
+				AM.reset_plane_and_layer()
```

Comment:
Is this actually correct? If I delete an unrelated structure on a turf where a movable is offset for some other unrelated reason, will it retain its state? For example, consider maploaded wallshifted objs.

Reply:
Wallshifted objs should be setting their default pixel offsets as well as their pixel offsets though.

Reply:
Do any actually do so (particularly mapped ones, but also there are some constructed ones and some that set offset in update_icon).

Reply:
Likely not, will need to go have a look.
Possibly structures/machines should set their default pixel offsets to whatever pixel_x/pixel_y are set to during Initialize()?

Reply:
I went ahead and made `/atom/Initialize()` set default pixel offsets if they're not set to anything, and to set pixel offsets otherwise. Between that and updating several of the hardcoded offset atoms like APCs to use default pixel offsets and reset_offsets() I think it should be fine now.